src/commands/createFunctionApp/FunctionAppCreateStep.ts

```diff
@@ -142,8 +153,8 @@ export class FunctionAppCreateStep extends AzureWizardExecuteStep<IFunctionAppWi
         const storageConnectionString: string = (await getStorageConnectionString(context)).connectionString;
         let appSettings: NameValuePair[] = [
             {
-                name: ConnectionKey.Storage,
-                value: storageConnectionString
+                name: `${ConnectionKey.Storage}__accountName`,
+                value: context.newStorageAccountName ?? context.storageAccount?.name
```

Comment:
> This is where a lot of our customer scenarios will break since we relied on that setting for a lot, so still plenty of work to do there.

Can you outline which scenarios you expect to break? Have you verified them breaking?